tchannel/zipkin/formatters.py

```diff
@@ -124,12 +129,15 @@ def binary_annotation_formatter(annotation):
     if isinstance(value, unicode):
         value = value.encode('utf-8')
 
-    return ttypes.BinaryAnnotation(
+    ba = ttypes.BinaryAnnotation(
         key=annotation.name,
-        stringValue=value,
-        annotationType=annotation_type
+        annotationType=annotation_type,
     )
 
+    setattr(ba, annotation.annotation_type+'Value', value)
```

Comment:
```
struct BinaryAnnotation {
    1: string key
    2: optional string stringValue
    3: optional double doubleValue
    4: optional bool boolValue
    5: optional binary bytesValue
    6: optional i64 intValue
    7: AnnotationType annotationType
}
```

the values in binary annotations are named by the pattern like type+'Value'.  So instead of lots of `if-else`, we can just use this way to set value of the binary annotation.